File: 2025-04-06/Container_With_Most_Water.java
package amazon;

public class Container_With_Most_Water {
    public static int maxArea(int[] height) {
        int left=0, right=height.length-1;
        int maxArea=0;

        while(left<right) {
            int length = right-left;
            int breath = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, length*breath);

            if(height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxArea;
    }
    public static void main(String[] args) {
        int[] arr = {1,8,6,2,5,4,8,3,7};
        System.out.println(maxArea(arr));
    }
}

File: 2026-01-26/Container_With_Most_Water.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        if not height:
            return 0

        max_area = 0
        l, r = 0, len(height)-1
        while l < r:
            length = r - l
            breadth = min(height[l], height[r])
            max_area = max(max_area, length * breadth)

            if height[l] < height[r]:
                l += 1
            elif height[l] > height[r]:
                r -= 1
            else:
                l += 1
                r -= 1

        return max_area


sol = Solution()
print(sol.maxArea([8,7,2,1]))